string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,85 @@
+// first i need to find each word
+// they will always start the beginning of the string or after a space or after a hyphen
+// they will always end at a space or at the end of the string or before a hyphen
+// i need an empty string to send each word to that i can check against the minor words list and check the length of
+// Rules: Capitalize the first word of the title and of any subtitle
+//          always capitalize title[0]
+//          title[0].toUpperCase(), start loop at title[1]
+//          capitalize if a first letter is two indexes after a colon
+//          if (title[i - 2] === ':') {storageString[0].toUpperCase()}
+// Capitalize all major words, including the second part of hyphenated words
+//          capitalize if a first letter follows a hyphen
+//          if (title[i - 1] === '-') {storageString[0].toUpperCase()}
+// Capitalize all words of four or more letters
+//          capitalize if the word is at least 4 letters
+//          if (storageString.length > 3) {storageString[0].toUpperCase()}
+// list of minor words: ['and', 'or', 'nor', 'but', 'a', 'an', 'the', 'as', 'at', 'by', 'for', 'in', 'of', 'on', 'per', 'to']
+// toLowerCase() word if it matches any word in this array
+//          for (var minorWordsIndex = 0; minorWordsIndex < minorWords.length; minorWordsIndex++)
+// hardcoded words: 'JavaScript', 'API'
+//          if (storageString.toLowerCase() === 'javascript') {storageString = 'JavaScript'}
+//          if (storageString.toLowerCase() === 'javascript:') {storageString = 'JavaScript:'}
+//          if (storageString.toLowerCase() === 'api') {storageString = 'API'}
+
+// create new string for final output
+// create a storage string that will hold each word i find
+// loop over the original string
+//    add each character to the storage string
+//    if character is a space or a hyphen or im at the end of title, start running my checks on the storage string
+//        if (title[i] === ' ' || title[i] === '-' || i = title.length - 1)
+//    capitalize according to any applicable rules
+//    add storage string to final output string
+//    clear the storage string
+// return the final output
+//    finalOutput += storageString;
+//    storageString = '';
+
+function titleCase(title) {
+  var finalOutput = '';
+  var storageString = '';
+  var minorWords = ['and ', 'or ', 'nor ', 'but ', 'a ', 'an ', 'the ', 'as ', 'at ', 'by ', 'for ', 'in ', 'of ', 'on ', 'per ', 'to '];
+  var firstLetterIndex = 0;
+  var firstChar = '';
+  var otherChars = '';
+  for (var i = 0; i < title.length; i++) {
+    storageString += title[i];
+    if (title[i] === ' ' || title[i] === '-' || i === title.length - 1) {
+      if (storageString.toLowerCase() === 'javascript ') {
+        storageString = 'JavaScript ';
+      } else if (storageString.toLowerCase() === 'javascript: ') {
+        storageString = 'JavaScript: ';
+      } else if (storageString.toLowerCase() === 'api') {
+        storageString = 'API';
```

Comment:
Why do your conditions for the variants of "javascript" include a space at the end, but this one doesn't? How does API avoid getting a space added? Is this dependent on the specific placement of the word API?